collection-scripts/gather_monitoring

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+BASE_COLLECTION_PATH="/must-gather"
+
+function collect() {
+  local resources target 
+  resources=()
+  target="$1"
+
+  mkdir -p "${BASE_COLLECTION_PATH}/${target}"
+  for i in $(/usr/bin/oc get "${target}.monitoring.coreos.com" --all-namespaces | grep "openshift-" | awk '{ print $1 }')
+  do
+    resources+=("$i")
+  done
+  
+  # we use nested loops to nicely output objects partitioned per namespace, kind
+  for resource in ${resources[@]}; do 
+    echo "INFO: Gathering ServiceMonitors in ${resource} namespace"
+    /usr/bin/oc get "${target}.servicemonitors.monitoring.coreos.com" -n "${resource}" -o yaml > "${BASE_COLLECTION_PATH}/${target}/${resource}.yaml"
```

Comment:
Should we be collecting this with a related resource?

Reply:
This is almost impossible to gather with related resource right now as related resource would need to be implemented in every operator managing namespace in which ServiceMonitor is deployed.

Reply:
I still think we should push this to related resource across this operators over a script like this. 

@deads2k do you agree?

Reply:
@sferich888 agreed, we'll implement related resources with respect to service monitors deployed in the `openshift-monitoring` namespace only. Please note that service monitors deployed by other operators are not going to be covered and must be implemented in their corresponding operator code.

We'll close this one out in favor of a PR against cluster-monitoring-operator setting those related fields for its own assets.